arroyo-df/src/types.rs

```diff
@@ -117,6 +89,65 @@ pub fn rust_to_arrow(typ: &Type) -> Option<NullableType> {
     }
 }
 
+pub fn data_type_to_arrow_type(data_type: &DataType) -> ArrowType {
+    let t = match data_type {
+        DataType::UInt64 => ArrowTypeEnum::Uint64(EmptyMessage {}),
+        DataType::Utf8 => ArrowTypeEnum::Utf8(EmptyMessage {}),
+        DataType::Null => ArrowTypeEnum::None(EmptyMessage {}),
+        DataType::Boolean => ArrowTypeEnum::Bool(EmptyMessage {}),
+        DataType::Int8 => ArrowTypeEnum::Int8(EmptyMessage {}),
+        DataType::Int16 => ArrowTypeEnum::Int16(EmptyMessage {}),
+        DataType::Int32 => ArrowTypeEnum::Int32(EmptyMessage {}),
+        DataType::Int64 => ArrowTypeEnum::Int64(EmptyMessage {}),
+        DataType::UInt8 => ArrowTypeEnum::Uint8(EmptyMessage {}),
+        DataType::UInt16 => ArrowTypeEnum::Uint16(EmptyMessage {}),
+        DataType::UInt32 => ArrowTypeEnum::Uint32(EmptyMessage {}),
+        DataType::Float16 => ArrowTypeEnum::Float16(EmptyMessage {}),
+        DataType::Float32 => ArrowTypeEnum::Float32(EmptyMessage {}),
+        DataType::Float64 => ArrowTypeEnum::Float64(EmptyMessage {}),
+        DataType::Date32 => ArrowTypeEnum::Date32(EmptyMessage {}),
+        DataType::Date64 => ArrowTypeEnum::Date64(EmptyMessage {}),
+        DataType::Binary => ArrowTypeEnum::Binary(EmptyMessage {}),
+        DataType::LargeBinary => ArrowTypeEnum::LargeBinary(EmptyMessage {}),
+        DataType::LargeUtf8 => ArrowTypeEnum::LargeUtf8(EmptyMessage {}),
+        _ => panic!("Unsupported DataType"),
```

Comment:
We shouldn't panic here unless we guarantee we can never hit this case

Reply:
I've made sure the planning surfaces the error for unsupported types.